engine/src/test/java/io/zeebe/engine/processor/workflow/instance/CreateWorkflowInstanceWithResultTest.java

```diff
@@ -48,64 +42,19 @@ public static void init() {
   public void shouldSendResultAfterCompletion() {
     // given
     Mockito.clearInvocations(ENGINE.getCommandResponseWriter());
-    final long workflowInstanceKey =
-        ENGINE
-            .workflowInstance()
-            .ofBpmnProcessId("WORKFLOW")
-            .withResult()
-            .withRequestId(1L)
-            .withRequestStreamId(1)
-            .create();
-
-    // then
-    final Record<WorkflowInstanceResultRecordValue> workflowInstanceResult =
-        RecordingExporter.workflowInstanceResultRecords()
-            .withWorkflowInstanceKey(workflowInstanceKey)
-            .withIntent(WorkflowInstanceResultIntent.COMPLETED)
-            .getFirst();
-
-    Assertions.assertThat(workflowInstanceResult.getValue())
-        .hasBpmnProcessId("WORKFLOW")
-        .hasVersion(1)
-        .hasWorkflowInstanceKey(workflowInstanceKey);
-
-    verify(ENGINE.getCommandResponseWriter(), times(1))
-        .intent(WorkflowInstanceResultIntent.COMPLETED);
-    verify(ENGINE.getCommandResponseWriter(), times(1)).tryWriteResponse(1, 1L);
-  }
-
-  @Test
-  public void shouldSendResultWithVariablesAfterCompletion() {
-    // given
-    Mockito.clearInvocations(ENGINE.getCommandResponseWriter());
-    final Map variables = Map.of("x", "foo");
-    final long workflowInstanceKey =
-        ENGINE
-            .workflowInstance()
-            .ofBpmnProcessId("WORKFLOW")
-            .withVariables(variables)
-            .withResult()
-            .withRequestId(2L)
-            .withRequestStreamId(2)
-            .create();
-
-    // when
-    final Record<WorkflowInstanceResultRecordValue> workflowInstanceResult =
-        RecordingExporter.workflowInstanceResultRecords()
-            .withWorkflowInstanceKey(workflowInstanceKey)
-            .withIntent(WorkflowInstanceResultIntent.COMPLETED)
-            .getFirst();
+    ENGINE
+        .workflowInstance()
+        .ofBpmnProcessId("WORKFLOW")
+        .withVariables(Map.of("x", "foo"))
+        .withResult()
+        .withRequestId(1L)
+        .withRequestStreamId(1)
+        .create();
 
     // then
-    Assertions.assertThat(workflowInstanceResult.getValue())
-        .hasBpmnProcessId("WORKFLOW")
-        .hasVersion(1)
-        .hasWorkflowInstanceKey(workflowInstanceKey);
-
-    assertThat(workflowInstanceResult.getValue().getVariables().get("x")).isEqualTo("foo");
-    verify(ENGINE.getCommandResponseWriter(), times(1))
+    verify(ENGINE.getCommandResponseWriter(), timeout(1000).times(1))
```

Comment:
could we intercept the call somehow such that we are able to verify that the right values are send ?

Reply:
Updated the test. Pls have a look.